frontend/src/metabase/visualizations/components/ChartSettings/ChartSettings.tsx

```diff
@@ -100,22 +135,35 @@ export const ChartSettings = ({
     return transformedSeries;
   }, [chartSettingsRawSeries]);
 
-  const widgets = useMemo(
-    () =>
-      propWidgets ||
-      getSettingsWidgetsForSeries(
-        transformedSeries,
-        handleChangeSettings,
-        isDashboard,
-        { dashboardId: dashboard?.id },
-      ),
-    [
-      propWidgets,
-      transformedSeries,
-      handleChangeSettings,
-      isDashboard,
-      dashboard?.id,
-    ],
+  return {
+    chartSettings,
+    handleChangeSettings,
+    chartSettingsRawSeries,
+    transformedSeries,
+  };
+};
+
+export const ChartSettings = ({
+  initial,
+  series,
+  computedSettings: propComputedSettings,
+  onChange,
+  question,
+  widgets,
+  chartSettings,
+  transformedSeries,
+}: ChartSettingsProps & UseChartSettingsStateReturned) => {
+  const [currentSection, setCurrentSection] = useState<string | null>(
+    initial?.section ?? null,
+  );
+  const [currentWidget, setCurrentWidget] = useState<Widget | null>(
+    initial?.widget ?? null,
+  );
+  const [popoverRef, setPopoverRef] = useState<HTMLElement | null>();
+
+  const computedSettings = useMemo(
+    () => propComputedSettings || {},
+    [propComputedSettings],
   );
```

Comment:
I wonder why we need `useMemo` here. Is this because the `{}` fallback value will cause the reference to change, and that causes re-rendering?

Reply:
Ok I had a deeper look at this one. It looks like that `computedSettings` will at _least_ be at least `{}` so this shouldn't actually be an issue. I can move this to a default prop and call it a day